hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/HapiTestEngine.java

```diff
@@ -201,72 +201,31 @@ private void appendTestsInClasspathRoot(
-            setParent(parent);
-
-            // Look for any methods supported by this class.
-            ReflectionUtils.findMethods(testClass, IS_HAPI_TEST, TOP_DOWN).stream()
-                    .filter(method -> {
-                        // The selectors tell me if some specific method was selected by the IDE or command line,
-                        // so I will filter out and only include test methods that were in the selectors, if there
-                        // are any such selectors. NOTE: We're not doing class selectors and such, a more robust
-                        // implementation probably should.
-                        final var selectors = discoveryRequest.getSelectorsByType(MethodSelector.class);
-                        for (final var selector : selectors) {
-                            if (!selector.getJavaMethod().equals(method)) {
-                                return false;
-                            }
-                        }
-                        return true;
-                    })
-                    .map(method -> new MethodTestDescriptor(method, this))
-                    .forEach(this::addChild);
-
-            // Skip construction of the Hedera instance if there are no test methods
-            skip = getChildren().isEmpty();
-        }
-
-        @Override
-        public Type getType() {
-            return Type.CONTAINER;
-        }
-
-        @Override
-        public SkipResult shouldBeSkipped(HapiTestEngineExecutionContext context) {
-            return skip ? SkipResult.skip("No test methods") : SkipResult.doNotSkip();
+        public HapiEngineDescriptor(UniqueId uniqueId) {
+            super(uniqueId, "Hapi Test");
         }
 
         @Override
         public HapiTestEngineExecutionContext before(HapiTestEngineExecutionContext context) {
             try {
-                // If we have a HapiTestSuite that is without tests we still want to show it as ignored,
-                // but we don't want to waste time setting up a node
-                if (allTestsSkipped()) {
-                    return new HapiTestEngineExecutionContext(null, null);
-                }
+                // Deleting the test data. Currently, we are deleting the data/saved and the eventstreams folders.
+                // We need to do that in order to be able to run all tests at the same time. Without that the tests
+                // are interfering with each other.
+                // Also, If we encounter a scenario where tests in the same suite are interfering with each other we
+                // can move this logic inside the after method in the MethodTestDescriptor class.
+                // This way we will clean up the data after each test.
+                if (context.getSavedStateDirectory() != null)
+                    FileUtils.deleteDirectory(context.getSavedStateDirectory());
+                if (context.getEventsLogDir() != null) FileUtils.deleteDirectory(context.getEventsLogDir());
```

Comment:
Moving the deleting of the temp folders in the `before` method instead of the `after` means we will always have the temp data from the last test suite present. Probably not a problem, I just wanted to point that out.

On the bright side - we will have the temp data from each test and it that might be helpful when debugging.

Reply:
Maybe we could do cleanup before and after? Dunno, mixed feelings on this

Reply:
Moving it first does solve one problem -- when you terminate the test early and the code to cleanup the directory doesn't run, then subsequent runs would fail without manual cleanup. This way, it works even in that case. And like @vtronkov said, we do have some stuff lying around afterward in case we need to look at it. I think this is probably better.